File: Assets/Scripts/Unused Scripts/ArcherAI.js


public var attackRange : float = 13f;
public var distance : float;
private var player : GameObject ;
public var attack : boolean = false;
private var destination: Transform;
private var agent : NavMeshAgent;
public var fireRate : float = 100f;
public var time : int = 90;
public var speed : float  = 1000;
public var launcher : Transform ;
public var projectile : GameObject;

function Start ()
{
	launcher = transform.FindChild ("launcher");
	player = GameObject.Find("Player");
	destination = player.transform;
	agent = gameObject.GetComponent.<NavMeshAgent>();
	
	agent.SetDestination(destination.position);
}
function Update () 
{	
	distance = Vector3.Distance(this.transform.position, player.transform.position);
	
	if(distance <= attackRange)
	{
		attack = true;
		agent.Stop();
	}else{
		attack = false;
		agent.Resume ();
		agent.SetDestination(destination.position);
	}
	if(distance <= attackRange)
	{
		attack = true;
	}else{
		attack = false;
	}
	
	if(time > 100 && attack)
	{		
		transform.LookAt(GameObject.FindGameObjectWithTag("Player").transform);
		var bullet : Rigidbody = Instantiate(launcher,projectile.transform.position, transform.rotation).GetComponent.<Rigidbody>();
		bullet.AddForce(transform.forward * speed);
		time = 0;
	}else if (attack){
		time++;
	}
}
function Dead(){
	enabled = false;
}

File: Assets/Scripts/Enemies/ArcherAI.cs
using UnityEngine;
using System.Collections;

public class ArcherAI : MonoBehaviour 
{
	public float attackRange = 13f;
	public float distance;
	private GameObject player;
	public bool attack  = false;
	private Transform destination;
	private NavMeshAgent agent;
	public float fireRate = 100f;
	private int time = 90;
	public float speed = 1000;
	//private Transform launcher;
	public GameObject projectile;

	void Start ()
	{
		//launcher = transform.FindChild ("launcher");
		player = GameObject.Find("Player");
		destination = player.transform;
		agent = gameObject.GetComponent<NavMeshAgent>();
		
		agent.SetDestination(destination.position);
	}
	void Update () 
	{	

		distance = Vector3.Distance(this.transform.position, player.transform.position);
		
		if(distance <= attackRange)
		{
			attack = true;
			agent.Stop();
		}else{
			attack = false;
			agent.Resume ();
			agent.SetDestination(destination.position);
		}

		if (attack && time >= fireRate) {
			attackPlayer ();
		} else if (attack) {
			time++;
		}
	}
	void Dead(){
		enabled = false;
	}

	void attackPlayer(){
		transform.LookAt(GameObject.FindGameObjectWithTag("Player").transform);
	//	bullet = Instantiate(projectile,launcher.transform.position, transform.rotation
		//bullet.AddForce(transform.forward * speed);
		time = 0;
	}

	
}